server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/XodusServerIssueStore.java

```diff
@@ -231,14 +233,15 @@ private static ServerHotspot adaptHotspot(Entity storedHotspot) {
     var endLine = (Integer) storedHotspot.getProperty(END_LINE_PROPERTY_NAME);
     var endLineOffset = (Integer) storedHotspot.getProperty(END_LINE_OFFSET_PROPERTY_NAME);
     var textRange = new org.sonarsource.sonarlint.core.commons.TextRange(startLine, startLineOffset, endLine, endLineOffset);
+    var vulnerabilityProbability = VulnerabilityProbability.valueOf((String) storedHotspot.getProperty(VULNERABILITY_PROBABILITY_PROPERTY_NAME));
     return new ServerHotspot(
       (String) requireNonNull(storedHotspot.getProperty(KEY_PROPERTY_NAME)),
       (String) requireNonNull(storedHotspot.getProperty(RULE_KEY_PROPERTY_NAME)),
       requireNonNull(storedHotspot.getBlobString(MESSAGE_BLOB_NAME)),
       filePath,
       textRange,
       (Instant) requireNonNull(storedHotspot.getProperty(CREATION_DATE_PROPERTY_NAME)),
-      Boolean.TRUE.equals(storedHotspot.getProperty(RESOLVED_PROPERTY_NAME)));
+      Boolean.TRUE.equals(storedHotspot.getProperty(RESOLVED_PROPERTY_NAME)), requireNonNull(vulnerabilityProbability));
```

Comment:
I don't think you need the requireNonNull here, the `valueOf` could throw earlier so it's not possible. And I don't think it's needed to catch on the valueOf, I don't see this case happening